VideoLocker/src/main/java/org/edx/mobile/view/UserProfileFragment.java

```diff
@@ -168,26 +171,25 @@ private void setAccount(@Nullable final Account account) {
                 viewHolder.locationContainer.setVisibility(View.GONE);
             } else {
                 viewHolder.limitedView.setVisibility(View.GONE);
-                if (account.getLanguageProficiencies().isEmpty()) {
-                    viewHolder.languageContainer.setVisibility(View.GONE);
-                } else {
-                    viewHolder.languageContainer.setVisibility(View.VISIBLE);
-                    viewHolder.languageText.setText(
-                            new Locale.Builder()
-                                    .setLanguage(account.getLanguageProficiencies().get(0).getCode())
-                                    .build()
-                                    .getDisplayName());
+                viewHolder.languageContainer.setVisibility(View.GONE);
+                if (!account.getLanguageProficiencies().isEmpty()) {
+                    try {
+                        viewHolder.languageText.setText(
+                                LocaleUtils.getLanguageNameFromCode(account.getLanguageProficiencies().get(0).getCode()));
```

Comment:
Questions:
1) Why are we only targeting the zeroth element of the `languageProficiencies` list ?
2) Why have a list when we only need to access only 1 element ?